### DISCUSSION TITLE
Is there Parquet Logical Types implementation?

### DISCUSSION THREAD

**Discussion by ftelnov** [apache/arrow-rs#5660] [Q&A] [answered]
As I can see for column [reader](https://docs.rs/parquet/latest/parquet/column/reader/enum.ColumnReader.html#variants), it is only possible to read physical data without its logical type representation. If I don't want to use arrow and want to simply use parquet, how can I read logical types then?

**Comment by tustvold** [comment, marked as answer]
You will need to interpret the logic type information encoded in the schema manually. I would strongly encourage using arrow, which handles this and other record shredding for you (among other things), it will be an easier experience and the resulting code will likely be faster

**Reply by ftelnov** [reply]
Thanks for the answer! Clarify a bit, please: 
Would it be possible to use arrow with any parquet file? Or, speaking differently, would writer side need to use arrow for serialization as well?

**Reply by tustvold** [reply]
You can read a parquet file written by a non-arrow writer, e.g. spark, with the arrow reader. It will map the parquet logical types as appropriate, certain specific things like timezone information won't roundtrip due to limitations in parquet's type system, but broadly speaking it should do what you would expect

**Reply by ftelnov** [reply]
Thanks a lot for the info, glad it works that way!
